File: src/main/java/org/shark/example/service/quartz/FindJobService.java
package org.shark.example.service.quartz;

import lombok.RequiredArgsConstructor;
import org.quartz.*;
import org.quartz.impl.matchers.GroupMatcher;
import org.shark.example.service.base.pojo.ResponseDto;
import org.shark.example.service.quartz.pojo.JobDto;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@RequiredArgsConstructor
@Service
public class FindJobService {

    private final Scheduler scheduler;

    public ResponseDto<List<JobDto>> start() {
        GroupMatcher<JobKey> matcher = GroupMatcher.anyJobGroup();
        List<JobDto> jobList = new ArrayList<>();
        try {
            Set<JobKey> jobKeys = scheduler.getJobKeys(matcher);
            for (JobKey jobKey : jobKeys) {
                List<? extends Trigger> triggers = scheduler.getTriggersOfJob(jobKey);
                for (Trigger trigger : triggers) {
                    JobDto jobDetails = new JobDto();
                    if (trigger instanceof CronTrigger) {
                        CronTrigger cronTrigger = (CronTrigger) trigger;
                        jobDetails.setCronExpression(cronTrigger.getCronExpression());
                        jobDetails.setTimeZone(cronTrigger.getTimeZone().getDisplayName());
                    }
                    jobDetails.setTriggerGroupName(trigger.getKey().getName());
                    jobDetails.setTriggerName(trigger.getKey().getGroup());
                    jobDetails.setJobGroupName(jobKey.getGroup());
                    jobDetails.setJobName(jobKey.getName());
                    jobDetails.setStartTime(trigger.getStartTime());
                    jobDetails.setJobClassName(scheduler.getJobDetail(jobKey).getJobClass().getName());
                    jobDetails.setNextFireTime(trigger.getNextFireTime());
                    jobDetails.setPreviousFireTime(trigger.getPreviousFireTime());
                    jobDetails.setStatus(scheduler.getTriggerState(trigger.getKey()).name());
                    jobList.add(jobDetails);
                }
            }
        } catch (SchedulerException e) {
            throw new RuntimeException(e);
        }

        return ResponseDto.<List<JobDto>>builder().status(1).build();
    }
}

File: src/main/kotlin/org/shark/example/service/quartz/FindJobService.kt
package org.shark.example.service.quartz

import org.quartz.*
import org.quartz.impl.matchers.GroupMatcher
import org.shark.example.service.base.pojo.ResponseDto
import org.shark.example.service.quartz.pojo.JobDto
import org.springframework.stereotype.Service
import java.util.ArrayList

@Service
class FindJobService(
    private val scheduler: Scheduler
) {
    fun start(): ResponseDto<List<JobDto>> {
        val matcher = GroupMatcher.anyJobGroup()
        val jobList = ArrayList<JobDto>()
        try {
            val jobKeys = scheduler.getJobKeys(matcher)
            for (jobKey in jobKeys) {
                val triggers = scheduler.getTriggersOfJob(jobKey)
                for (trigger in triggers) {
                    val jobDetails = JobDto()
                    if (trigger is CronTrigger) {
                        jobDetails.cronExpression = trigger.cronExpression
                        jobDetails.timeZone = trigger.timeZone.displayName
                    }
                    jobDetails.triggerGroupName = trigger.key.name
                    jobDetails.triggerName = trigger.key.group
                    jobDetails.jobGroupName = jobKey.group
                    jobDetails.jobName = jobKey.name
                    jobDetails.startTime = trigger.startTime
                    jobDetails.jobClassName = scheduler.getJobDetail(jobKey).jobClass.name
                    jobDetails.nextFireTime = trigger.nextFireTime
                    jobDetails.previousFireTime = trigger.previousFireTime
                    jobDetails.status = scheduler.getTriggerState(trigger.key).name
                    jobList.add(jobDetails)
                }
            }
        } catch (e: SchedulerException) {
            throw RuntimeException(e)
        }

        return ResponseDto.builder<List<JobDto>>().status(true).data(jobList).build()
    }
}
